solid/solid-file-client/solid.js

```diff
@@ -226,31 +210,31 @@ function getSolidDocumentUrl(resourceUrl) {
 }
 
 async function fetchUserProfile(webId) {
-    const profileQuads = await readSolidDocument(webId);
-    const nameQuad = profileQuads.find(quad => quad.predicate.value === 'http://xmlns.com/foaf/0.1/name');
-    const storageQuad = profileQuads.find(quad => quad.predicate.value === 'http://www.w3.org/ns/pim/space#storage');
-
+    const [nameQuad] = await readSolidDocument(webId, null, { foaf: 'name' })
+    await findPodStorage(webId)
     return {
         url: webId,
-        name: nameQuad?.object.value || 'Anonymous',
-        storageUrl: storageQuad?.object.value || await findUserStorage(webId),
+        name: nameQuad?.object.value || webId.split('//')[1].split('/profile')[0], // replace 'Anonymous',
+        storageUrl: await findPodStorage(webId),
     };
 }
 
-async function findUserStorage(url) {
+async function findPodStorage(url) {
     url = url.replace(/#.*$/, '');
     url = url.endsWith('/') ? url + '../' : url + '/../';
     url = new URL(url);
 
-    const response = await solidFileClient.fetch(url.href);
+    // following solid/protocol used by NSS and CSS
+    const [StorageQuad] = await readSolidDocument(url.href, null, null, { space: 'Storage' })
```

Comment:
This will only exist on a profile document, otherwise we should look at the header like the previous code was doing and as described [in the spec](https://solidproject.org/TR/protocol#storage). So I don't think this should happen here, I would keep in the `findPodStorage` method.

Reply:
- No this is not the profile document `storage`, but the `Storage` that appears in a folder index.ttl. 
```
n0:
    a ldp:BasicContainer, ldp:Container, space:Storage;
```
This is recursive until a `Storage` is found`

- The problem with the header is that NSS actually do not have access to the https://podName `container headers` when there is an index.html at root, but only to `resource headers`. A very old issue, may be resolved with https://github.com/solid/node-solid-server/pull/1642

Reply:
I see, in that case why not check for both? Either the header or the `a space:Storage` triple. In that case, I'd also mention the same comment I made about looking for the task, we're missing the `http://www.w3.org/1999/02/22-rdf-syntax-ns#type` predicate in the filter.